### PR TITLE
Cleaning up JArchiveGzip.

### PR DESCRIPTION
**Pull request by Mathewlenning**
Did the same refactoring as Bzip2.
1. Isolated JError into temporary method
2. separated old extraction logic from streaming logic.

### CODE REVIEW

File: libraries/joomla/archive/gzip.php [hunk 1]
```diff
@@ -58,124 +58,107 @@ public function extract($archive, $destination, array $options = array ())
...
+		}
 
-			while ($this->_data);
+		while ($this->_data);
 
-			$output->close();
-			$input->close();
+		$output->close();
+		$input->close();
+		return true;
```
Comment (wilsonge) [inline libraries/joomla/archive/gzip.php]: I know travis isn't erroring on this (need to investigate why) but code style rules are a new line before a return

File: libraries/joomla/archive/gzip.php [hunk 2]
```diff
@@ -58,124 +58,107 @@ public function extract($archive, $destination, array $options = array ())
...
 
-				if ($this->_data)
+			return $this->raiseWarning(100, 'Unable to write archive (gz)');
+		}
+
+		do
+		{
+			$this->_data = $input->read($input->get('chunksize', 8196));
+
+			if ($this->_data)
```
Comment (wilsonge) [inline libraries/joomla/archive/gzip.php]: Can we not merge this if statement and the one beneath it together?

**Comment by wilsonge** [comment]
Looks good from a code review generally :)

File: libraries/joomla/archive/gzip.php (hunk 2 above)
Comment (Mathewlenning) [inline libraries/joomla/archive/gzip.php]: @wilsonge honestly I don't know, but the best way to find out is to give it a try right!  Nice catch =^D

**Comment by Kubik-Rubik** [comment]
Also looks good to me. We won't add this in the 3.4.1 but we can merge it for the 3.4.2 version. Thank you @Mathewlenning!

**Comment by Mathewlenning** [comment]
Sounds good =^D Hopefully I'll have a few more ready by then.

Sincerely,
Mathew Lenning

Babel-University.com <http://babel-university.com/>
406-626-3251
LinkedIn Profile <http://linkedin.com/pub/mathew-lenning/30/268/52b/>

**Comment by Hackwar** [comment]
I brought this up on the mailinglist a few weeks ago: I think this is a good example for a refactoring that belongs int a minor release and not in a patch release. Likewise with the other refactoring that you did.

**Comment by Mathewlenning** [comment]
@hannes

You'll hear no complaints from me. 3.4.x or 3.5.0 either way works. And if we are going full SemVer then 3.5 would be the best place for it. Imagine how much we could clean the core up by then. 

Sincerely,
Mathew Lenning

Babel-university.com

P.S. This message was sent via iPhone, so please forgive any errors

**Comment by Kubik-Rubik** [comment]
@Mathewlenning @Hackwar Okay, good point. I'll move the milestone to 3.5. Thank you for your input!

**Comment by Mathewlenning** [comment]
I was just wondering what is the RTC tag for?

**Comment by mbabker** [comment]
Ready to commit

**Comment by Mathewlenning** [comment]
Thanks =^D Still learning the lingo

Sincerely,
Mathew Lenning

Babel-University.com <http://babel-university.com/>
406-626-3251
LinkedIn Profile <http://linkedin.com/pub/mathew-lenning/30/268/52b/>

**Comment by wilsonge** [comment]
Merged into 3.5 dev branch. Thanks Mathew!

**Pull request closed without merging**